tests/beta/tapes/test_cv_param_shift.py

```diff
@@ -0,0 +1,830 @@
+            2 * r0
+        ) * np.sinh(2 * r1)
+        expected[3] = 0.5 * np.sin(phi0 - phi1) * np.sinh(2 * r0) * np.sinh(2 * r1)
+
+        assert np.allclose(grad_A2, expected, atol=tol, rtol=0)
+
+    cv_ops = [getattr(qml, name) for name in qml.ops._cv__ops__]
+    analytic_cv_ops = [cls for cls in cv_ops if cls.supports_parameter_shift]
+
+    @pytest.mark.parametrize("obs", [qml.X, qml.P, qml.NumberOperator, qml.Identity])
+    @pytest.mark.parametrize("op", analytic_cv_ops)
+    def test_gradients_gaussian_circuit(self, op, obs, mocker, tol):
+        """Tests that the gradients of circuits of gaussian gates match between the
+        finite difference and analytic methods."""
+        tol = 1e-2
+
+        args = np.linspace(0.2, 0.5, op.num_params)
+
+        with CVParamShiftTape() as tape:
+            qml.Displacement(0.5, 0, wires=0)
+            op(*args, wires=range(op.num_wires))
+            qml.Beamsplitter(1.3, -2.3, wires=[0, 1])
+            qml.Displacement(-0.5, 0.1, wires=0)
+            qml.Squeezing(0.5, -1.5, wires=0)
+            qml.Rotation(-1.1, wires=0)
+            expval(obs(wires=0))
+
+        dev = qml.device("default.gaussian", wires=2)
+        res = tape.execute(dev)
+
+        tape._update_gradient_info()
+        tape.trainable_params = set(range(2, 2 + op.num_params))
+
+        # check that every parameter is analytic
+        for i in range(op.num_params):
+            assert tape._par_info[2 + i]["grad_method"][0] == "A"
+
+        spy = mocker.spy(CVParamShiftTape, "parameter_shift_first_order")
+        grad_F = tape.jacobian(dev, method="numeric")
+        grad_A2 = tape.jacobian(dev, method="analytic", force_order2=True)
+
+        spy.assert_not_called()
+        assert np.allclose(grad_A2, grad_F, atol=tol, rtol=0)
+
+        if obs.ev_order == 1:
+            grad_A = tape.jacobian(dev, method="analytic")
+            spy.assert_called()
+            assert np.allclose(grad_A, grad_F, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("t", [0, 1])
+    def test_interferometer(self, t, tol):
+        """An integration test for CV gates that support analytic differentiation
+        if succeeding the gate to be differentiated, but cannot be differentiated
+        themselves (for example, they may be Gaussian but accept no parameters,
+        or may accept a numerical array parameter.).
+
+        This ensures that, assuming their _heisenberg_rep is defined, the quantum
+        gradient analytic method can still be used, and returns the correct result.
+
+        Currently, the only such operation is qml.Interferometer. In the future,
+        we may consider adding a qml.GaussianTransfom operator.
+        """
+
+        if t == 1:
+            pytest.xfail(
+                "There is a bug in the second order CV parameter-shift rule; "
+                "phase arguments return the incorrect derivative."

```

Comment:
@nquesada, I was wondering if you have any ideas to fix this? I think you might be the person who knows the logic best